### PR TITLE
Fix some scrollbar track and border painting issues

### LINKED ISSUE #90310: Some Scrollbar track and border painting issues
#### Code
```dart
import 'package:flutter/material.dart';
import 'package:flutter/gestures.dart';

main() => runApp(MaterialApp(
  theme: ThemeData().copyWith(
    scrollbarTheme: _scrollbarTheme(),
  ),
  scrollBehavior: MyCustomScrollBehavior(),
  home: MyApp(),
));

class MyApp extends StatefulWidget {
  @override
  MyAppState createState() => MyAppState();
}

class MyAppState extends State<MyApp> {
  ScrollController verticalController = ScrollController();
  ScrollController horizontalController = ScrollController();
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('ScrollBar')),
      body: Container(
        child: Scrollbar(
          notificationPredicate: (n) => n.depth == 1,
          // thickness: 10,
          // hoverThickness: 50,
          scrollbarOrientation: ScrollbarOrientation.left,
          // showTrack: true,
          showTrackOnHover: true,
          isAlwaysShown: true,
          controller: verticalController,
          child: Scrollbar(
            scrollbarOrientation: ScrollbarOrientation.top,
            isAlwaysShown: true,
            controller: horizontalController,
            child: SingleChildScrollView(
              controller: horizontalController,
              scrollDirection: Axis.horizontal,
              physics: const AlwaysScrollableScrollPhysics(),
              child: SingleChildScrollView(
                controller: verticalController,
                physics: const AlwaysScrollableScrollPhysics(),
                child: Column(
                  children: [
                    Row(
                      children: [
                        FlutterLogo(
                          size: 1200,
                        ),
                      ],
                    ),
                  ],
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class MyCustomScrollBehavior extends MaterialScrollBehavior {
  // Override behavior methods and getters like dragDevices
  @override
  Set<PointerDeviceKind> get dragDevices => {
    PointerDeviceKind.touch,
    PointerDeviceKind.mouse,
    // etc.
  };

  @override
  Widget buildScrollbar(BuildContext context, Widget child, ScrollableDetails details) {
    // When modifying this function, consider modifying the implementation in
    // the base class as well.
    return child;
  }
}


ScrollbarThemeData _scrollbarTheme({
  MaterialStateProperty<double?>? thickness,
  // bool showTrack = true,
  bool showTrackOnHover = true,
  bool isAlwaysShown = true,
  Radius radius = const Radius.circular(0.0),
  MaterialStateProperty<Color?>? thumbColor,
  MaterialStateProperty<Color?>? trackColor,
  MaterialStateProperty<Color?>? trackBorderColor,
  double crossAxisMargin = 3.0,
  double mainAxisMargin = 50.0,
  double minThumbLength = 30.0,
}) {
  return ScrollbarThemeData(
    thickness: thickness ?? MaterialStateProperty.resolveWith(_getThickness),
    // showTrack: showTrack,
    showTrackOnHover: showTrackOnHover,
    isAlwaysShown: isAlwaysShown,
    radius: radius,
    thumbColor: thumbColor ?? MaterialStateProperty.resolveWith(_getThumbColor),
    trackColor: trackColor ?? MaterialStateProperty.resolveWith(_getTrackColor),
    trackBorderColor: trackBorderColor ?? MaterialStateProperty.resolveWith(_getTrackBorderColor),
    crossAxisMargin: crossAxisMargin,
    mainAxisMargin: mainAxisMargin,
    minThumbLength: minThumbLength,
  );
}

double? _getThickness(Set<MaterialState> states) {
  if (states.contains(MaterialState.hovered))
    return 20.0;
  return 10.0;
}

Color? _getThumbColor(Set<MaterialState> states) {
  return null;
  if (states.contains(MaterialState.dragged))
    return Colors.red;
  if (states.contains(MaterialState.hovered))
    return Colors.blue;
  return Colors.green;
}

Color? _getTrackColor(Set<MaterialState> states) {
  if (states.contains(MaterialState.hovered))
    return Colors.yellowAccent;
  return null;
}

Color? _getTrackBorderColor(Set<MaterialState> states) {
  if (states.contains(MaterialState.hovered))
    return Colors.purple;
  return null;
}
```
### Expect
image
### Actually
image

### PR DESCRIPTION
**Pull request by xu-baolin**
Fixes #90310

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -437,38 +437,55 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
...
         y = size.height - thickness - crossAxisMargin - padding.bottom;
-        trackOffset = Offset(0.0, y - crossAxisMargin);
+        trackOffset = Offset(mainAxisMargin, y - crossAxisMargin);
         break;
     }
 
     _trackRect = trackOffset & trackSize;
+    Offset borderStart;
+    Offset borderEnd;
+    switch(resolvedOrientation) {
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollbar.dart:468]: Can these not be included in the switch statements above, rather than repeating?

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

Thank you for the updates! And apologies for the merge conflicts. 😅

**Comment by zanderso** [comment]
Why did the github bot land this while the tree is red? @godofredoc @dnfield

**Comment by dnfield** [comment]
image

You can see here that when it landed, the check was marked as passing.

In the past we've had issues where the API we used had limits around how many times we could change the status - I don't know if such a limit exists with the current API, but @keyonghan or @CaseyHillers might know.

**Comment by dnfield** [comment]
@christopherfujino suggested having the bot check LUCI specifically before merging and not relying on the status on the PR, which might be a good idea. We should fix the status anyway though since it'd be confusing to humans who are wondering why the bot isn't merging their PR.

**Pull request merged**